api/test/wfl/tools/parallel_runner.clj

```diff
@@ -0,0 +1,22 @@
+(ns wfl.tools.parallel-runner
+  (:require [clojure.test :as test]))
+
+(defn -main
+  [& args]
+  (letfn [(go! [& tests]
+            (let [output (new java.io.StringWriter)]
+              (binding [test/*test-out* output
+                        test/*report-counters* (ref test/*initial-report-counters*)]
+                (doseq [t tests] (t))
+                [@test/*report-counters* (str output)])))]
+    (let [ns      (symbol (first args))
+          _       (use ns)
+          tests   (group-by (comp #(contains? % :parallel) meta)
+                            (filter (comp :test meta) (vals (ns-interns (the-ns ns)))))
+          futures (doall (map #(future (go! %)) (tests true)))
+          results (apply list (apply go! (tests false)) (map deref futures))
+          summary (assoc (apply merge-with + (map first results))
+                    :type :summary)]
+      (run! println (filter not-empty (map second results)))
+      (test/do-report summary)
+      (System/exit (if (test/successful? summary) 0 1)))))
```

Comment:
I couldn't find any easy docs on how to properly return an exit code and literally just returning 0 or 1 didn't actually seem to work for some reason